### ISSUE TITLE
Add a `custom_vjp` for `ratintn` — it's a linear operator, so its VJP is its transpose

### ISSUE THREAD

**Issue by joglekara** [ergodicio/tsadar#121]
## Summary

`ratintn` (`ratintn.py`) is **exactly linear in its first argument `f`** — verified numerically at max rel err 1.7e-12, with `f(0) == 0` exactly. So its VJP is simply its transpose, and both directions share the same `g`-only coefficients.

Today reverse-mode differentiates through the whole body, including the double-`where` guard at `ratintn.py:48-59`. That guard exists *specifically* because AD evaluates both branches — from the comment in the source:

> `jnp.where` picks `rfn` for the value, but autodiff differentiates both branches and propagates `0*inf = nan`. On CPU `gav` is a tiny non-zero so the `rf` gradient is finite (`0*huge=0`); on GPU FMA/rounding lands `gav` at exactly 0 -> inf -> nan.

A `custom_vjp` sidesteps this entirely: compute the coefficients once from `g`, apply them forward in the primal and transposed in the backward pass. The unselected branch is never differentiated, so the `gav_safe` workaround stops being load-bearing for correctness on GPU.

## Structure to exploit

`ratcen` factors as `out = P * f_av + Q * f_dif`, where `P` and `Q` depend only on `g`:

```
rf  = fav*(1/gav_s + gdif**2/(12*gav_s**3)) + fdif*(-gav*gdif/(12*gav_s**3))
rfn = fav*(L/gdif)                          + fdif*(1/gdif - gav*L/gdif**2)
```

with `L = log((gav + 0.5j*gdif)/(gav - 0.5*gdif))`. Since `f_av` and `f_dif` are two-point stencils, the forward pass is a weighted sum and the backward pass is the adjoint of that same stencil — no complex `log` in the backward pass at all, and residuals shrink to the coefficient arrays instead of every intermediate.

## Scope and expected payoff

Two call sites, and they benefit differently:

- **1D path** (`form_factor.py:280-282`): `g` is a compile-time constant, so the whole block collapses to a fixed matrix. That is strictly better than a `custom_vjp` and is tracked separately in #118 — **do that one instead**, this issue does not apply there.
- **ARTS/2D path** (`form_factor.py:444`): `g = vx - xie_mag_at` varies per evaluation point, so no single matrix exists. This is where the `custom_vjp` pays: the per-point coefficient row is computed once and reused between primal and cotangent.

**Sequencing:** `ratintn` is only ~0.2% of ARTS cost as currently written (4 us/point out of 2266) — `rotate` dominates completely. This is worth doing for the GPU-NaN robustness on its own merits, but the *performance* benefit only becomes measurable after #120 lands.

## Notes

- Should be a behavior-preserving change; worth a test asserting the `custom_vjp` gradient matches the current autodiff gradient on inputs away from poles, plus one near a pole (`gdif` close to `1e-4 * gav`) where the branch selection flips.
- Benchmarked on CPU; the NaN failure mode described in the source comment is GPU-specific, so the robustness claim needs a GPU run to confirm.
